### ISSUE TITLE
The application not retrieving the strings from StringPacks.

### ISSUE THREAD

**Issue by 02gaurav** [WhatsApp/StringPacks#7]
I have many modules in my application. I have done all the steps mentioned.
Let me explain each part :
 1. `StringPackIds` is part of the app module (root module)
 2. I have copied the [scripts/](https://github.com/WhatsApp/StringPacks/blob/main/library/scripts) and [pack.gradle](https://github.com/WhatsApp/StringPacks/blob/main/library/pack.gradle) to project source dir. Also  `config.json` is also kept on the same place 
 3. Also mapping and `.pack` are generated properly in module where i have all strings file (including translated one)
 4. No while i am setting string to a textview from xml or programmeticly like `android:text="@string/some_thing"` ,
  Its not translated even after i set the locale correctly. If there is any other doc or any tutorial let me know but it seems by just doing these steps it does not work

Edited : - How we will know if string is packable or not?

**Comment by ankit-gautam23** [comment]
@02gaurav could you plz assign it to me?

**Comment by 02gaurav** [comment]
@ankit-gautam23 I don't have permission to assign

**Comment by ashish29dec** [comment]
Could you please send me the link to your git where you are seeing the issue? Also, are you building StringPacks code on your own or using Maven's version?

**Comment by ashish29dec** [comment]
@02gaurav if you can send me the link to your github where you are facing the issue I can try looking into what exactly is happening because it seems to be working in my testing.

**Comment by 02gaurav** [comment]
@ashish29dec I have set it up again. Now it's working only for a few strings. It's creating .pack files and string-pack dir. But only for a few strings, even though I have translations for all the strings. Can you tell me how we will know if the string is packable or not?

**Comment by ashish29dec** [comment]
@02gaurav Yes not all strings get stringpacked. Some strings and hence their translations are needed even before `StringPacks` is initialized.
To know which strings are stringpacked, you can take a look at `StringPackIds.java`. It lists all strings that are packed in the pack file

**Comment by 02gaurav** [comment]
@ashish29dec Which types of strings cannot be packed? Even when similar strings are packed with different words, the same string cannot be packed. How can I determine if a particular string can be packed or not? If there is a way to tell if a string is not packable just by looking at it, then I can rephrase it to make it packable. This would be beneficial in packing more strings.

**Comment by ashish29dec** [comment]
@02gaurav To address your initial question.
> No while i am setting string to a textview from xml or programmeticly like android:text="@string/some_thing" ,
Its not translated even after i set the locale correctly. If there is any other doc or any tutorial let me know but it seems by just doing these steps it does not work

Are you able to see translations for this string after enabling `StringPacks` in your code? Even if a certain string's translation is not included in the pack file it should still show the correct translation when running the app in that locale.
If `StringPacks` does not find a translation in `.pack` file, it falls back to using the one from `values-xx/strings.xml`
So, I would like to know if you are able to see the translation for all the strings (packed or not packed)?

**Comment by 02gaurav** [comment]
@ashish29dec I am to see around 50% of strings in the string pack.

**Comment by ashish29dec** [comment]
@02gaurav it is ok for some strings to not get stringpacked and it is expected behavior. However, regardless of whether strings are stringpacked or not you should always see correct translations when running the app in a certain locale.
If you could share a sample github link where you are able to reproduce the issue of not all translations showing up in the app, maybe I can help you debug this better.

**Comment by 02gaurav** [comment]
@ashish29dec That is private GitHub repo. I can not share the link. Let me set up a new project and try the same in that project.

**Comment by ashish29dec** [comment]
@02gaurav any updates on the sample project showing the issue?

**Comment by ashish29dec** [comment]
Closing this Issue as no updates. Please open a new Issue if you are still facing a problem.
